src/west/manifest.py

```diff
@@ -313,32 +324,38 @@ class Project:
     __slots__ = 'name remote url path abspath clone_depth revision'.split()
 
     def __init__(self, name, remote, defaults, path=None, clone_depth=None,
-                 revision=None, url=None):
+                 revision=None):
         '''Specify a Project by name, Remote, and optional information.
 
-        :param name: Project's user-defined name in the manifest
-        :param remote: Remote instance corresponding to this Project's remote.
-                       This may not be None.
+        :param name: Project's user-defined name in the manifest.
+        :param remote: Remote instance corresponding to this Project as
+                       specified in the manifest. This is used to build
+                       the project's URL, and is also stored as an attribute.
+        :param defaults: If the revision parameter is not given, the project's
+                         revision is set to defaults.revision.
         :param path: Relative path to the project in the west
-                     installation, if present in the manifest. If None,
+                     installation, if present in the manifest. If not given,
                      the project's ``name`` is used.
+        :param clone_depth: Nonnegative integer clone depth if present in
+                            the manifest.
         :param revision: Project revision as given in the manifest, if present.
+                         If not given, defaults.revision is used instead.
         '''
         if remote is None:
             raise ValueError('remote or url must be given')
```

Comment:
Maybe now that your add it, revert this comment back to:
`raise ValueError('remote or url must be given')`
as `url` is not valid anymore. (leftover from before we made SpecialProject)

Reply:
Good catch, thank you!